### ISSUE TITLE
recovering from Windows reinstall; howto

### ISSUE THREAD

**Issue by bbulkow** [microsoft/WSL#5416]
**Is your feature request related to a problem? Please describe.**
This request is not related to a problem; per se.

**Describe the solution you'd like**
I recently had a fault in the underlying parts of windows, and needed to go through a recovery seqeunce. In that sequence, I chose "restore my data files". My data files - including all the directories in the windows file system which I use with WSL1 - are perfectly intact. I suspect that, somwhere in my machine, my WSL root points are 

However, my WSL mappings are lost. 

Could you please publish instructions for this exact case? How to find your WSL workspaces, where they tend to reside (appdata?) and how to map them into distributions again?

I did take a backup, and have a zip file that is 6 months old. Since I don't keep my data in WSL (thankfully I don't use WSL2), the older version is a good starting point, but it's especially frustrating as I know the entire up-to-date distribution is very likely somewhere on my drive, taking up space, and I am, at least theoretically, one choice command away from reviving.

**Describe alternatives you've considered**
I have searched the drive for file names and patterns. No dice.
I have considered scanning the source for the wsl command tools, which specifically included the --export command to see from what directories it pushes files.

**Additional context**
I do take backups --- of my host OS. I have most of my up-to-date data files. The issue is that any program that forces me to take extra "export" steps to separately backup, creates a fragility.

**Comment by bbulkow** [comment]
Yes, this is a question. I currently have a laptop that _very probably_ had a key environment and I can't find it. I wish I'd get some help, just the name of one directory would be a massive, massive boon.

**Comment by onomatopellan** [comment]
Open %LocalAppData%/Packages and if your distro is Ubuntu you will find a Canonical* folder. Inside there is a LocalState folder and you should find the root structure there.

**Comment by licanhua** [comment]
There are two piece of information you need to take care of:
1. rootfs in LocalState, just like @onomatopellan  said, here is  the folder for Ubuntu20.04. `%LocalAppData%/Packages\CanonicalGroupLimited.Ubuntu20.04onWindows_79rhkp1fndgsc\LocalState`
2. `HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\CurrentVersion\Lxss`

WSL also provides --export and --import which helps you to backup and restore the distros.

**Comment by licanhua** [comment]
Personally I prefer to make backup/restore as a new feature.
--export and --import did part of job, but it's for advanced user only. 
There are some restrictions for --export and --import:
1. User need to remember distro names and locations, otherwise, when restore the distro, the shortcut like 'Ubuntu' will not work anymore.
2. If settings in regkey is missing, distro may re-initialize and override the existing file system

I would like the command is simplified to:
1. `wsl --backup [all or distro name] [tar file name]`
2. 'wsl --restore [tar file name]'

**Comment by bbulkow** [comment]
It's somewhat worse, because there is no way to expose the location. At least with distros you can get the list with --list. It would seem adding the feature to get the disto location _after the install_ is necessary.

Feature request?
